### DISCUSSION TITLE
JTS 1.20 precision rounding ?

### DISCUSSION THREAD

**Discussion by ebocher** [locationtech/jts#1074] [General]
JTS update  breaks some unit tests in H2GIS due to precision rounding.

Here's an example using the buffer operation.

```java
WKTReader wktReader = new WKTReader();
Geometry geom = wktReader.read("POINT(100 90)");
BufferOp bufOp  = new BufferOp(geom, new BufferParameters(2));
bufOp.getResultGeometry(50);
```

It returns

```
POLYGON ((150 90, 135.35533905932738 54.64466094067263, 100 40, 64.64466094067262 54.64466094067262, 50 90, 64.64466094067262 125.35533905932738, 99.99999999999999 140, 135.35533905932738 125.35533905932738, 150 90))
```

previous (JTS 1.19) 

```
POLYGON ((150 90, 135.35533905932738 54.64466094067263, 100 40, 64.64466094067262 54.64466094067262, 50 90, 64.64466094067262 125.35533905932738, 100 140, 135.35533905932738 125.35533905932738, 150 90))
```

Is this normal?

**Comment by dr-jts** [comment]
Hmm, that's not good.  It would be better to provide precise results in cases like this, where the result should be exact.

I'm not sure why the numeric calculation is not being rounded.  I'm not aware of any code change which would cause this (but there might be something with unanticipated side effects).  We should look into this.

**Comment by dr-jts** [comment]
On my platform (JRE 17.0.7) this test produces the exact value (i.e. identical to JTS 1.19).  I wonder if this is a JVM version issue?

**Comment by ebocher** [comment]
This is a funny thing: H2GIS tests are running now.
Maybe a cache issue with my IDE!
Many thanks for the new version of JTS
We will be able to extend the production of Local Climate Zones with [GeoCLIMATE](https://github.com/orbisgis/geoclimate)
